crates/iceberg-catalog/src/service/storage/s3.rs

```diff
@@ -1181,7 +1184,7 @@ fn validate_bucket_name(bucket: &str) -> Result<(), ValidationError> {
     Ok(())
 }
 
-fn insert_pyiceberg_hack(config: &mut TableProperties) {
+fn push_fsspec_fileio_with_s3v4restsigner(config: &mut TableProperties) {
```

Comment:
pyiceberg_hack was a bit more telling about why we're doing this, maybe add a small comment?

Reply:
I thought `pyiceberg_hack ` was an aweful name. It didn't say anything about what it was doing and why either. Just that it was for Pyiceberg. Added a comment now.